py/dml/traits.py

```diff
@@ -568,8 +607,7 @@ def __init__(self, ancestors, ancestor_vtables):
         # vtable the method or parameter appears
         self.ancestor_vtables = ancestor_vtables
 
-    @property
-    def ancestors(self): return self.ancestry_paths.keys()
+        self.ancestors = set(self.ancestry_paths.keys())
```

Comment:
This change is purely to avoid `merge_method_impl_maps()` from being pathologically `O(n^4)` in behavior. That ordo was too high for me to be comfortable.

Reply:
really? I'm pretty sure `dict.keys()` is a lazy O(1) view of the key set (class `dict_keys`), offering O(1) `in` and O(n) iteration. Unlike `set`, it also preserves iteration order.

Reply:
Oh hey, you're right! I tried to research about this and drew the conclusion that `dict_keys` was just an iterator, but I've experimentally it does seem to be `O(1)`. Clearly my research failed.

Reply:
Reverted.